middleman-core/lib/middleman-core/util/data.rb

```diff
@@ -43,7 +43,7 @@ def parse(full_path, known_type=nil)
     end
 
     case [start, stop]
-    when %w(--- ---), %w(--- ...)
+    when %w(--- ---), %w(--- ...), %(\\--- \\---), %w(\\--- ...)
       [parse_yaml(frontmatter, full_path), additional_content]
```

Comment:
Not necessary, the `frontmatter` variable already has the delimiters stripped away.